### ISSUE TITLE
Building a statically linked libyaml into PyYAML wheel

### ISSUE THREAD

**Issue by shakefu** [yaml/pyyaml#303]
I'm trying to build a PyYAML wheel that leverages libyaml without having to install libyaml as a system shared library for all the services we use. We do have control of the runtime environment (via containers) so we are able to target a fixed version of libyaml/gcc/etc.

Is there any way using the existing build toolchain to set it up so that libyaml should be statically linked instead of dynamically linked and included with the C extension distribution in the binary wheel?

**Comment by ssbarnea** [comment]
I totally susbscribe to this as the warning which I seen recently worries me:
```
WARNING: PyYAML will be slow due to installation without libyaml support for interpreter:
```
It should be possible to build binary wheels for MacOS and Linux (manylinux). 

Packaging binary would be huge benefits for lots of users.

**Comment by MRigal** [comment]
See also #227

**Comment by etanol** [comment]
Does this also include `setup.py` modifications to be able to build the extension **statically linked** to a custom built `libyaml`?  Or should I open a different issue?

In my case, I can't find ways to control `CFLAGS`, `LDFLAGS` _et al_.  Neither from `pip` nor from `python setup.py`

**Comment by nitzmahone** [comment]
addressed in #407 in 5.4b1
